taichi/program/program_impl.cpp

```diff
@@ -10,4 +10,24 @@ void ProgramImpl::compile_snode_tree_types(SNodeTree *tree) {
   TI_NOT_IMPLEMENTED;
 }
 
+void ProgramImpl::dump_cache_data_to_disk() {
+  auto &mgr = get_kernel_compilation_manager();
+  mgr.clean_offline_cache(offline_cache::string_to_clean_cache_policy(
+                              config->offline_cache_cleaning_policy),
+                          config->offline_cache_max_size_of_files,
+                          config->offline_cache_cleaning_factor);
+  mgr.dump();
+}
+
+KernelCompilationManager &ProgramImpl::get_kernel_compilation_manager() {
+  if (kernel_com_mgr_) {
```

Comment:
OOC: why do we only construct it when dumping cache_data_to_disk?

Reply:
the function is called in:
* `ProgramImpl::dump_cache_data_to_disk()`
* `XXXProgramImpl::compile()`

next PRs, the `AotModuleBuilder`s will also use the `KernelCompilationManager`